src/coinbase/coinbase.ts

```diff
@@ -77,7 +77,7 @@ export class Coinbase {
     useServerSigner = false,
     debugging = false,
     basePath = BASE_PATH,
-  }: CoinbaseOptions) {
+  }: CoinbaseOptions = {}) {
```

Comment:
that's great. I think we don't need to set empty string for `apiKeyName` and `privateKey`. let's make sure `CoinbaseOptions` is also updated: 

```
  /**
   * The API key name.
   */
  apiKeyName: string;

  /**
   * The private key associated with the API key.
   */
  privateKey: string;


```

Reply:
If these are required fields than the deconstructor has to have default values for them So we can't really get around the empty string. 

Alternatively, if they are optional on the type and we do not provide the empty string default value -> we must do both undefined and empty string check which seems less clean

Reply:
I tested the changes locally and they worked for me. I removed the `?` mark from `apiKeyName` and `privateKey`, and also removed the default empty strings and default value for `CoinbaseOptions`. Let's sync on this to discuss further.